src/main.h

```diff
@@ -289,6 +289,7 @@ bool IsInitialBlockDownload();
  * This function only returns the highest priority warning of the set selected by strFor.
  */
 std::string GetWarnings(const std::string& strFor);
+std::string GetWarnings(const std::string& strFor, bool verbose);
```

Comment:
I'd consider changing the variable name to something more descriptive like fForStaking. The warnings returned when the flag is false are not less verbose than the staking-related.